### PR TITLE
A few more type hints

### PR DESCRIPTION
**Pull request by alexrudd2**
- More type hints ( I had them lying around my computer and forgot)
- Shorter Exception [syntax](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement)

### CODE REVIEW

File: pymodbus/events.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def __init__(self, **kwargs):
         self.listen = kwargs.get("listen", False)
         self.broadcast = kwargs.get("broadcast", False)
 
-    def encode(self):
+    def encode(self) -> str:
```
Comment (janiversen) [inline pymodbus/events.py:57]: I think a message is bytes ?

File: pymodbus/events.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def encode(self):
         packet = pack_bitstring(bits)
         return packet
 
-    def decode(self, event):
+    def decode(self, event) -> None:
```
Comment (janiversen) [inline pymodbus/events.py:67]: Would be nice to type event as well (as bytes).

File: pymodbus/framer/base.py [hunk 3]
```diff
@@ -18,15 +19,16 @@ class ModbusFramer:
 
     name = ""
 
-    def __init__(self, decoder, client=None):
+    def __init__(self, decoder, client=None) -> None:
```
Comment (janiversen) [inline pymodbus/framer/base.py:22]: Would be nice to type decoder and client as well.

**Review by janiversen — COMMENTED** [review]
I have just made a couple of comments, you decide if you want to do them.

File: pymodbus/events.py (hunk 1 above)
Comment (alexrudd2) [reply pymodbus/events.py:57]: Thanks, you are correct.  I'll annotate `pack_bitstring` and improve its docstring.

https://github.com/pymodbus-dev/pymodbus/blob/9c6acc385d2cb5362863cad740a90f693785f1e0/pymodbus/utilities.py#L109-L119

Comment (alexrudd2) [reply pymodbus/events.py]: Fixed in f5cd9fa817a5ec4fe19281c079d45e5307c3484e.

I think b212e79faf94379f59a0fb429230064d8c212ef6 is technically an API change (changing the name of an argument), although all current callers pass by value.

File: pymodbus/framer/base.py (hunk 3 above)
Comment (alexrudd2) [reply pymodbus/framer/base.py]: decoder I could do in f7b7517bc9dc5a3df5b01220ecb95b331fa579c6, client creates a circular import :(

File: pymodbus/events.py (hunk 1 above)
Comment (janiversen) [reply pymodbus/events.py]: you are right technically It is, but there are no changes in the app, so lets not mark it.

API_Changes.rst signals changes the user need to consider/modify (at least that is the intention).

**Pull request merged**